src/terms/kinetic.jl

```diff
@@ -1,22 +1,66 @@
 """
-Kinetic energy: 1/2 sum_n f_n ∫ |∇ψn|^2.
+Kinetic energy: 1/2 sum_n f_n ∫ |∇ψn|^2 (standard)
+             or 1/2 sum_n f_n ∫ blowup(-i∇Ψ)^2 (modified kinetic term).
 """
-struct Kinetic
-    scaling_factor::Real
+Base.@kwdef struct Kinetic{F}
+    scaling_factor::Real = 1
+    blowup::F = BlowupKineticEnergy() # blow-up function to smooth energy bands. Defaut is x↦x^2.
 end
-Kinetic(; scaling_factor=1) = Kinetic(scaling_factor)
-(kin::Kinetic)(basis) = TermKinetic(basis, kin.scaling_factor)
+
+(kin::Kinetic)(basis) = TermKinetic(basis, kin.scaling_factor, kin.blowup)
 function Base.show(io::IO, kin::Kinetic)
     fac = isone(kin.scaling_factor) ? "" : ", scaling_factor=$scaling_factor"
     print(io, "Kinetic($fac)")
 end
 
+"""
+Define different energy cutoff strategies.
+ADD REF when the paper is published.
+"""
+struct BlowupKineticEnergy end
+struct BlowupCHV end
+
+"""
+Default blow-up, x->x^2, corresponding to the standard kinetic energies.
+"""
+(blowup::BlowupKineticEnergy)(x) = sum(abs2, x)
+
+"""
+Blow-up function as proposed in [REF paper Cancès, Hassan, Vidal to be submitted]
+Parameters have been computed to ensure C^2 regularity of the energies bands
+away from crossings and Lipschitz continuity at crossings.
+TODO: find a better C^2 interpolation.
+"""
+function (blowup::BlowupCHV)(x)
```

Comment:
I think I would just have it be a multiplicative thing, ie the kinetic energy is g^2/2 * blowup(g)